Services/LayoutModelManager.php

```diff
@@ -120,20 +123,56 @@ protected function getUniqueId($layout)
         return $layout->getId();
     }
 
-    protected function saveInDb($label, $twigPath, $previewPath, $orientation)
+    protected function saveInDb($path, $label, $previewPath, $orientation, $templates)
     {
         // Do not change the name if we update the layout
         if (null === $this->layout->getLabel()) {
             $this->layout->setLabel($label);
         }
-        $this->layout->setPath($twigPath);
         $this->layout->setPreviewPath($previewPath);
         $this->layout->setOrientation($orientation);
         $this->layout->setNotesModes(array(0 => 1));
         $this->layout->setCssVersion(1);
         $this->layout->setUpdated(new \DateTime());
 
-        $this->om->flush($this->layout);
+        // Deleting unused templates
+        foreach ($this->layout->getTemplates() as $template)
+        {
+            if (!array_key_exists($template->getType(), $templates))
+            {
+                $this->layout->removeTemplate($template);
+                $this->om->remove($template);
+            }
+        }
+
+        // Adding/updating templates
+        foreach ($templates as $templateType => $templateConfig)
+        {
+            if (!in_array($templateType, $this->layout->getTemplatesTypes()))
+            {
+                $template = new Template();
+
+                $template->setType($templateType);
+                $template->setPath($path.$templateConfig['file']);
+
+                $this->om->persist($template);
+                $this->layout->addTemplate($template);
+            }
+            else
+            {
+                $template = $this->layout->getTemplate($templateType);
+
+                if ($template->getPath() != $path.$templateConfig['file'])
+                {
+                    $template->setPath($path.$templateConfig['file']);
```

Comment:
See with @dvdn
When we upload a model we have a dependence with the layout id in the field path. 
Example: we have "uploads/3/stop.html.twig" & "uploads/3/line.html.twig"
We think it will be better to remove $path from it.
What do you think ?